### ISSUE TITLE
Following annotation on shape's member does not work

### ISSUE THREAD

**Issue by daddykotex** [awslabs/smithy-language-server#42]
TLDR: CMD + click on an annotation on a structure's member will not be followed

See before:

https://user-images.githubusercontent.com/5230460/168094104-5821dac0-40e2-4228-8aaa-e1dad6f40002.mov

See after: (this is based on a build at commit: e82fbffb1a1de7bd1bba0e3dd378c5bba46e9e76. this is not the commit that introduced the issue, it's just a reference that you can use if you want to reproduce.)

https://user-images.githubusercontent.com/5230460/168094213-5822ceb4-8241-4020-916f-06d2adc6a55f.mov

In the first video, we can see that CMD + click on an annotation on a structure's member will be followed (`@required` for example).

In the second video, we see that clicking on such an annotation does not work anymore. Clicking on annotation on top level shapes will work fine.

I don't have the bandwidth to troubleshoot the issue right now. I will come back to it later.
